src/card.py

```diff
@@ -77,53 +67,64 @@ def __str__(self) -> str:
         """
         return self.to_str(False)
 
+    def __repr__(self) -> str:
+        """
+        Default string creation
+        No emoji
+        """
+        return "card.Card<" + self.to_str(False) + ">"
+
+    def __hash__(self) -> int:
+        return hash(self.__str__())
+
     def __eq__(self, other: Any) -> bool:
         if not isinstance(other, Card):
             return False
         return (self.val == other.val) and (self.suit == other.suit)
 
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, Card):
+            raise TypeError(
+                f"'<' not supported between instance of 'Card' and '{type(other)}'"
+            )
+
+        if self.val == other.val:
+            return self.suit < other.suit
+
+        return self.val < other.val
+
     def __ne__(self, other: Any) -> bool:
         return not self.__eq__(other)
 
 
 def convert_card_array_to_enum_array(
-    cards: List[Card],
-) -> Tuple[List[CardVal], List[CardSuit]]:
+    cards: set[Card],
+) -> tuple[list[CardVal], list[CardSuit]]:
     """
     Function to turn an array of cards into it's components parts.
     I.e. a list of values, and a list of suits.
     """
     list_val = []
     list_suit = []
-    for i_card in cards:
+    for i_card in sorted(cards):
         list_val.append(i_card.val)
         list_suit.append(i_card.suit)
 
     return list_val, list_suit
```

Comment:
In retrospect, we don't need the _cards_ in order, we need the outputs in order.
```suggestion
    for i_card in cards:
        list_val.append(i_card.val)
        list_suit.append(i_card.suit)

    return sorted(list_val), sorted(list_suit)
```